src/lower.c

```diff
@@ -1525,31 +1526,70 @@ static int32_t lower_toggle_tag_for(type_t type, tree_t where, ident_t prefix, i
       flags = COV_FLAG_TOGGLE_PORT;
 
    if (type_is_array(type)) {
-      tree_t r = range_of(type, dims - 1);
-      int32_t first_tag = 0;
+      int t_dims = dimension_of(type);
+      tree_t r = range_of(type, t_dims - curr_dim);
+      int32_t first_tag = -1;
       int64_t low, high;
 
       if (folded_bounds(r, &low, &high)) {
          assert(low <= high);
-         for (int64_t i = low; i <= high; i++) {
+
+         int64_t i;
+         int64_t last;
+         int64_t first;
+         int inc;
+
+         if (cover_skip_array_toggle(cover_tags, high - low + 1))
+            return -1;
+         cover_add_dim(cover_tags);
```

Comment:
I'm not sure what `cover_add_dim` / `cover_sub_dim` do exactly. Does it just end up the same as the `curr_dim` argument to this function?

Reply:
Not really, `curr_dim` tracks current dimension in case of multi-dimensional array.
Dimension within `cover_tags`, tracks also dimension in case of nested array. Dimension
from nested array is not really held anywhere, since that is "array of array of array of...".
So, I need to track the information that I am currently already in second level.

Reply:
So that I can properly detect something as in `cover2` test.

Reply:
OK, but it's confusing that "dimension" is used to refer to both the actual array dimension and the nesting level of arrays in arrays. Can you call the later "depth" or "nesting" or something like that? E.g. `cover_inc_nesting()`, `cover_dec_nesting()`.